packages/procaptcha/src/modules/ProsopoCaptchaApi.ts

```diff
@@ -60,6 +60,21 @@ export class ProsopoCaptchaApi {
         try {
             const captchaChallenge = await this.providerApi.getCaptchaChallenge(this.userAccount, this.provider)
             this.verifyCaptchaChallengeContent(this.provider, captchaChallenge)
+            // convert https/http to match page
+            captchaChallenge.captchas.forEach((captcha) => {
+                captcha.captcha.items.forEach((item) => {
+                    if (item.data) {
+                        // drop the 'http(s):' prefix, leaving '//'. The '//' will autodetect http/https from the page load type
+                        // https://stackoverflow.com/a/18320348/7215926
+                        if (item.data.startsWith('https://')) {
```

Comment:
Any reason not to use a regex?

```
item.data = item.data.replace(/^http(s)*:\/\//, '')
```

And should we be supporting `file://` [et al.](https://en.wikipedia.org/wiki/List_of_URI_schemes)?

Reply:
regex might replace any http(s) string in another part of the url, e.g. a hash, e.g. 123abc1http23 becomes 123abc123

file:// still works with this mechanism, it leaves it untouched

Reply:
But the hat at the beginning of the regex command makes sure it's the start of the string. `^h`

Reply:
good point, I hadn't spotted the hat! will update :)

regarding the `file://`, I don't believe that triggers a mixed-content warning as the warning is only concerned about using http/https requests when the page is loaded in the other manner. `file://` is local and has no comparison to http(s) because it isn't encrypted whatsoever, doesn't need to be, so it _shouldnt_ trigger the warning. But I'll check :)

Reply:
only affects http(s), not other protocols :)